rest/route/agent.go

```diff
@@ -12,31 +12,58 @@ import (
 // GET /rest/v2/agent/cedar_config
 
 type agentCedarConfig struct {
-	settings *evergreen.Settings
+	config evergreen.CedarConfig
 }
 
-func makeAgentCedarConfig(settings *evergreen.Settings) gimlet.RouteHandler {
+func makeAgentCedarConfig(config evergreen.CedarConfig) *agentCedarConfig {
 	return &agentCedarConfig{
-		settings: settings,
+		config: config,
 	}
 }
 
 func (h *agentCedarConfig) Factory() gimlet.RouteHandler {
 	return &agentCedarConfig{
-		settings: h.settings,
+		config: h.config,
 	}
 }
 
-func (h *agentCedarConfig) Parse(ctx context.Context, r *http.Request) error {
-	return nil
-}
+func (*agentCedarConfig) Parse(_ context.Context, _ *http.Request) error { return nil }
 
 func (h *agentCedarConfig) Run(ctx context.Context) gimlet.Responder {
-	data := apimodels.CedarConfig{
-		BaseURL:  h.settings.Cedar.BaseURL,
-		RPCPort:  h.settings.Cedar.RPCPort,
-		Username: h.settings.Cedar.User,
-		APIKey:   h.settings.Cedar.APIKey,
+	return gimlet.NewJSONResponse(apimodels.CedarConfig{
+		BaseURL:  h.config.BaseURL,
+		RPCPort:  h.config.RPCPort,
+		Username: h.config.User,
+		APIKey:   h.config.APIKey,
+	})
+}
+
+// GET /rest/v2/agent/data_pipes_config
+
+type agentDataPipesConfig struct {
+	config evergreen.DataPipesConfig
+}
+
+func makeAgentDataPipesConfig(config evergreen.DataPipesConfig) *agentDataPipesConfig {
+	return &agentDataPipesConfig{
+		config: config,
 	}
-	return gimlet.NewJSONResponse(data)
+}
+
+func (h *agentDataPipesConfig) Factory() gimlet.RouteHandler {
+	return &agentDataPipesConfig{
+		config: h.config,
+	}
+}
+
+func (*agentDataPipesConfig) Parse(_ context.Context, _ *http.Request) error { return nil }
+
+func (h *agentDataPipesConfig) Run(ctx context.Context) gimlet.Responder {
+	return gimlet.NewJSONResponse(apimodels.DataPipesConfig{
+		Host:         h.config.Host,
+		Region:       h.config.Region,
+		AWSAccessKey: h.config.AWSAccessKey,
+		AWSSecretKey: h.config.AWSSecretKey,
```

Comment:
This is probably beyond this change, but it'd be nice if there was some way we could use the machine IAM roles for this rather than passing around keys.

Reply:
We discussed this a bunch, it isn't really feasible since we have static hosts outside of AWS.

Reply:
Cool, thanks for the info!